src/pages/DashboardPage/DashboardPage.test.tsx

```diff
@@ -0,0 +1,93 @@
+import { ThemeProvider } from '@mui/material';
+import { render, screen } from '@testing-library/react';
+import { describe, it, expect, vi, beforeAll, afterAll } from 'vitest';
+
+import { theme } from '../../style-library';
+
+import DashboardPage from './DashboardPage';
+
+// Mocks de componentes
+vi.mock('../../components/LeaderboardPage/LeaderboardPage', () => ({
+  LeaderboardPage: () => <div>Mock Leaderboard</div>,
+}));
+
+vi.mock('../../components/AvatarMenu/AvatarMenu', () => ({
+  AvatarMenu: () => <div>Mock AvatarMenu</div>,
+}));
+
+vi.mock('../../components/Logo/Logo', () => ({
+  Logo: () => <div>Mock Logo</div>,
+}));
+
+vi.mock('../../components/StatsCard/StatsCard', () => ({
+  StatsCard: ({ label, value }: { label: string; value: string | number }) => (
+    <div>
+      Mock StatsCard {label} {value}
+    </div>
+  ),
+}));
+
+vi.mock('../../components/TestCard/TestCard', () => ({
+  TestCard: ({ title }: { title: string }) => <div>Mock TestCard {title}</div>,
+}));
+
+// Guardar original para restaurar
+let originalMatchMedia: typeof window.matchMedia;
```

Comment:
You store originalMatchMedia as typeof window.matchMedia. In some test environments window.matchMedia may be undefined; this may cause type issues or runtime errors in some environments.

**Sugerencia:**
```typescript
Guard for undefined (e.g., const originalMatchMedia = window.matchMedia ?? undefined) or use vi.spyOn(window, 'matchMedia') and restore it with vi.restoreAllMocks() to ensure robust behavior across environments.
```